dev/update_changelog.py

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+# -*- coding: utf-8 -*-
+"""Script for automatically updating the `CHANGELOG.md` based on the commits since the latest release tag."""
+from pathlib import Path
+import re
+import subprocess
+
+DEFAULT_CHANGELOG_SECTIONS = """
+### ‼️ Breaking changes
+
+
+### ✨ New features
+
+
+### 🗑️ Deprecation
+
+
+### 👌 Improvements
+
+
+### 🐛 Bug Fixes
+
+
+### 📚 Documentation
+
+
+### 🔧 Maintenance
+
+
+### ⬆️ Update dependencies
+
+
+### ♻️ Refactor
+
+"""
+
+
+def update_changelog():
+    """Update the `CHANGELOG.md` for a first draft of the release."""
+
+    print('🔍 Checking the current version number')
+    with Path('CHANGELOG.md').open('r', encoding='utf8') as handle:
+        current_changelog = handle.read()
+
+    with Path('src/aiida_quantumespresso/__init__.py').open('r', encoding='utf8') as handle:
+        version = re.search(r"__version__ = '(?P<version_number>\d+.+)'", handle.read()).groupdict()['version_number']
+
+    if str(version) in current_changelog:
+        print('🛑 Current version already in `CHANGELOG.md`. Skipping...')
+        return
+
+    print('⬆️ Found updated version number, adapting `CHANGELOG.md`.')
+    tags = subprocess.run(['git', 'tag'], capture_output=True, check=False).stdout
```

Comment:
`pylint` wants you to explicitly set `check`:

https://pylint.pycqa.org/en/latest/user_guide/messages/warning/subprocess-run-check.html

So I added it quickly, since I wasn't really concerned with it, but happy to change it to `True`. This is actually exactly why `pylint` complains, so you think about it. I was just stubborn in my ignorance because I wanted to test the GitHub action ^^